go-controller/pkg/kubevirt/router.go

```diff
@@ -0,0 +1,274 @@
+	return DeleteRoutingForMigratedPodWithZone(nbClient, pod, nil)
+}
+
+// EnsureLocalZonePodAddressesToNodeRoute will add static routes and policies to ovn_cluster_route logical router
+// to ensure VM traffic work as expected after live migration if the pod is running at the local/global zone.
+//
+// Following is the list of NB logical resources created depending if it's interconnected or not:
+//
+// IC:
+//   - static route with cluster wide CIDR as src-ip prefix and nexthop GR, it has less
+//     priority than route to use overlay in case of pod to pod communication
+//
+// NO IC:
+//   - low priority policy with src VM ip and reroute GR, since it has low priority
+//     it will not override the policy to enroute pod to pod traffic using overlay
+//
+// Both:
+//   - static route with VM ip as dst-ip prefix and output port the LRP pointing to the VM's node switch
+func EnsureLocalZonePodAddressesToNodeRoute(watchFactory *factory.WatchFactory, nbClient libovsdbclient.Client, switchNames SwitchNames, pod *corev1.Pod) error {
+	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, types.DefaultNetworkName)
+	if err != nil {
+		return fmt.Errorf("failed reading ovn annotation: %v", err)
+	}
+
+	// VM is running at the node that owns the subnet the point to point
+	// routing is not needed
+	if switchNames.Current == switchNames.Original {
+		if err := DeleteRoutingForMigratedPod(nbClient, pod); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale static routes or policies for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		return nil
+	}
+	lrpName := types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + pod.Spec.NodeName
+	lrpAddresses, err := util.GetLRPAddrs(nbClient, lrpName)
+	if err != nil {
+		return fmt.Errorf("failed configuring pod routing when reading LRP %s addresses: %v", lrpName, err)
+	}
+
+	// For interconnect at static route with a cluster-wide src-ip address is
+	// needed to route egress n/s traffic
+	if config.OVNKubernetesFeature.EnableInterconnect {
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
+			// Policy to with low priority to route traffic to the gateway
+			ipFamily := utilnet.IPFamilyOfCIDR(clusterSubnet.CIDR)
+			nodeGRAddress, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, lrpAddresses)
+			if err != nil {
+				return err
+			}
+
+			egressStaticRoute := nbdb.LogicalRouterStaticRoute{
+				IPPrefix: clusterSubnet.CIDR.String(),
+				Nexthop:  nodeGRAddress.IP.String(),
+				Policy:   &nbdb.LogicalRouterStaticRoutePolicySrcIP,
+			}
+			if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(nbClient, types.OVNClusterRouter, &egressStaticRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+				return item.IPPrefix == egressStaticRoute.IPPrefix && item.Nexthop == egressStaticRoute.Nexthop && item.Policy != nil && *item.Policy == *egressStaticRoute.Policy
```

Comment:
You need to check if item.IPPrefix is a subnet of egressStaticRoute.IPPrefix

Reminder, the cluster subnet can be reconfigured to a bigger subnet